torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -309,9 +309,28 @@ void cacheAllocatorDeregisterHook(
   }
 }
 
+#ifdef NCCL_COMM_DUMP
```

Comment:
original proposal was to use the existing ifdef for NCCL_EXP, I am OK with this too if we think its worth having separate control.  Maybe if we think this could be upstreamed to nccl it makes sense to keep separate.  I think for convenience you could at least define this inside an ifdef NCCL_EXP block.  (check the correct name for it)